### PR TITLE
ADDS THE LOUD MOUTH TRAIT

### PR DESCRIPTION
**Pull request by Raitononai**
## About the PR

ADDS A NEW TRAIT THAT MAKES YOU YELL ALL THE TIME. 
IT MAKES YOU ALWAYS SPEAK IN CAPS AND APPENDS "!!" TO EVERYTHING WITH A 20% CHANCE TO MAKE YOU VOCALISE YOUR RAGE. RAAAAAARGH.

## Why / Balance

ITS FUNNY AND I'M ANGRY, GRAAAAAAH!

## Technical details

ITS JUST ACCENT CODE BUT IT CAPITALISES EVERYTHING AND APPENDS "!!" ITS NOT THAT CONFUSING.
ADDS "LoudSystem" AND "LoudAccentComponent". YOU COULD USE THEM ON SOMETHING LIKE A FUNNY DRINK OR CHEM EFFECT IF YOU WANT, JUST LIKE OTHER ACCENT COMPONENTS.

## Media

https://github.com/space-wizards/space-station-14/assets/136900466/cea3d43d-bd63-4c18-826f-6fa28b14962c

image

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: Added the new 'Loud Mouth' trait, it's now possible to really express your emotions. Permanently.

### CODE REVIEW

**Comment by Equivocateur** [comment]
Dr ANIMAL rp

**Comment by Tunguso4ka** [comment]
:trollface:

**Comment by deltanedas** [comment]
make it invert case so if you use caps it turns into lowercase
like that time we made a rage cage but called it calm cage and were screaming about GET ROBUST AT THE calm cage TODAY!!!!

**Comment by Raitononai** [comment]
I like this idea, i'll do it tomorrow

File: Content.Server/Speech/EntitySystems/LoudSystem.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<LoudAccentComponent, AccentGetEvent>(OnAccentGet);
+    }
+    public string Accentuate(string message, LoudAccentComponent component)
+    {
+
+        var msg = message;
```
Comment (ficcialfaint) [inline Content.Server/Speech/EntitySystems/LoudSystem.cs:21]:
```suggestion
    {
        var msg = message;
```

File: Content.Server/Speech/EntitySystems/LoudSystem.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            var pick = _random.Pick(component.YellSuffixes);
+
+            msg = msg + " " + Loc.GetString(pick);
+            return msg;
+        }
+
+
+
+ 
+        return msg;
```
Comment (ficcialfaint) [inline Content.Server/Speech/EntitySystems/LoudSystem.cs:45]: The comment says `Apply random suffix to var if random chance succeeds`, but apparently it's vice versa, isn't it? Also the if-statement should probably be inverted and there are some odd blank lines which should be removed.
```suggestion
        if (!_random.Prob(component.YellChance))
            return msg;

        var pick = _random.Pick(component.YellSuffixes);

        msg = msg + " " + Loc.GetString(pick);
        return msg;
```

**Review by ficcialfaint — CHANGES_REQUESTED** [review]

**Comment by Raitononai** [comment]
I decided against invert case cause its a bit confusing when you type in full caps and get lower case, with the trait that's meant to make you yell.

File: RobustToolbox
Comment (ficcialfaint) [inline RobustToolbox:1]: Your PR shouldn’t modify any of the submodules

**Review by ficcialfaint — CHANGES_REQUESTED** [review]

**Comment by Raitononai** [comment]
How the fuck did i manage to do this

**Comment by GoodWheatley** [comment]
> How the fuck did i manage to do this

Try updating the submodule on your local branch

**Comment by Raitononai** [comment]
i'm just gonna close this and re-make the PR since it's just faster for me to do that

**Pull request closed without merging**